src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php

```diff
@@ -138,20 +147,47 @@ public function deleteAction(int $stateId): RedirectResponse
      */
     public function editAction(int $stateId, Request $request): Response
     {
-        return $this->redirect(
-            $this->getAdminLink(
-                'AdminStates',
-                ['updatestate' => '', 'id_state' => $stateId],
-                true
-            )
-        );
+        try {
+            /** @var EditableState $editableState */
+            $editableState = $this->getQueryBus()->handle(new GetStateForEditing((int) $stateId));
+        } catch (StateException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_states_index');
+        }
+
+        $stateForm = null;
+
+        try {
+            $stateForm = $this->get('prestashop.core.form.identifiable_object.builder.state_form_builder')->getFormFor((int) $stateId);
+            $stateForm->handleRequest($request);
+            $result = $this->get('prestashop.core.form.identifiable_object.handler.state_form_handler')->handleFor((int) $stateId, $stateForm);
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_states_index');
+            }
+        } catch (StateNotFoundException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_states_index');
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render('@PrestaShop/Admin/Improve/International/Locations/State/edit.html.twig', [
+            'enableSidebar' => true,
+            'layoutTitle' => $this->trans('Edit: %value%', 'Admin.Actions', ['%value%' => $editableState->getName()]),
+            'stateForm' => $stateForm->createView(),
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+        ]);
     }
 
     /**
      * Show "Add new" form and handle form submit.
      *
      * @AdminSecurity(
-     *     "is_granted('create', request.get('_legacy_controller'))",
+     *     "is_granted(['create'], request.get('_legacy_controller'))",
```

Comment:
This is deprecated in symfony 4.4
```suggestion
     *     "is_granted('create', request.get('_legacy_controller'))",
```

Reply:
Fixed